src/Controller/ControllerFactory.php

```diff
@@ -249,6 +249,18 @@ protected function coerceStringToType(string $argument, ReflectionNamedType $typ
         switch ($type->getName()) {
             case 'string':
                 return $argument;
+            case 'float':
+                if (is_numeric($argument)) {
+                    return (float)$argument;
+                }
+                break;
+            case 'int':
+                if (is_numeric($argument)) {
```

Comment:
Couldn't the cast end up losing information here?  For example `12.1` passes is_numeric but dropping the `.1` isn't ideal. Should the int param type only accept integer strings? We could use `ctype_digit($argument)` or `preg_match('/^[0-9]+$/', $argument)` instead.

Reply:
The quick discussion we had was users should add regex to the route to match specific integers so we lazily cast when the route matches.

Do we help here or do we help in the route more?

Reply:
Routing patterns could be used to restrict what integer values are allowed, but I don't think we should be automatically rounding parameter values because the method has an int type hint. I know that is how PHP operates but I think we can be a bit more strict.

Reply:
Updated to validate int string.